microfactory/microfactory.go

```diff
@@ -199,6 +200,62 @@ func (p *GoPackage) FindDeps(config *Config, path string) error {
 	return nil
 }
 
+// Roughly equivalent to go/build.Context.match
+func matchBuildTag(name string) bool {
+	if name == "" {
+		return false
+	}
+	if i := strings.Index(name, ","); i >= 0 {
+		ok1 := matchBuildTag(name[:i])
+		ok2 := matchBuildTag(name[i+1:])
+		return ok1 && ok2
+	}
+	if strings.HasPrefix(name, "!!") {
+		return false
+	}
+	if strings.HasPrefix(name, "!") {
+		return len(name) > 1 && !matchBuildTag(name[1:])
+	}
+
+	if name == runtime.GOOS || name == runtime.GOARCH || name == "gc" {
+		return true
+	}
+	for _, tag := range build.Default.BuildTags {
+		if tag == name {
+			return true
+		}
+	}
+	for _, tag := range build.Default.ReleaseTags {
+		if tag == name {
+			return true
+		}
+	}
+
+	return false
+}
+
```

Comment:
How would you feel about a comment like "tells whether the given comment requests that the current file be included/excluded from compilation"?